x/marker/keeper/keeper.go

```diff
@@ -318,6 +318,29 @@ func (k Keeper) SetNetAssetValue(ctx sdk.Context, marker types.MarkerAccountI, n
 	return nil
 }
 
+// SetNetAssetValueWithBlockHeight adds/updates a net asset value to marker with a specific block height
+func (k Keeper) SetNetAssetValueWithBlockHeight(ctx sdk.Context, marker types.MarkerAccountI, netAssetValue types.NetAssetValue, source string, blockHeight uint64) error {
+	netAssetValue.UpdatedBlockHeight = blockHeight
+	if err := netAssetValue.Validate(); err != nil {
+		return err
+	}
+
+	setNetAssetValueEvent := types.NewEventSetNetAssetValue(marker.GetDenom(), netAssetValue.Price, netAssetValue.Volume, source)
+	if err := ctx.EventManager().EmitTypedEvent(setNetAssetValueEvent); err != nil {
+		return err
+	}
+
+	key := types.NetAssetValueKey(marker.GetAddress(), netAssetValue.Price.Denom)
+	bz, err := k.cdc.Marshal(&netAssetValue)
+	if err != nil {
+		return err
+	}
+	store := ctx.KVStore(k.storeKey)
+	store.Set(key, bz)
+
+	return nil
+}
```

Comment:
Looks good, but it looked really familiar. The function above it has exactly the same logic. Is it possible to update the `SetNetAssetValue` to use this so we don't have to maintain both?

Example...
```
func (k Keeper) SetNetAssetValue(ctx sdk.Context, marker types.MarkerAccountI, netAssetValue types.NetAssetValue, source string) error {
	return k.SetNetAssetValueWithBlockHeight(ctx, marker, netAssetValue, source, uint64(ctx.BlockHeight())
}
```

Reply:
I thought about that. However, @iramiller and I were talking about events and how they should work with the upgrade handler. I will wait till I hear back from my other conversation I added to the method for his input. Also, didn't know if it is something we might want to remove after upgrade.

Reply:
The larger flow for events feels like we would change the system to suppress "non-indexed" events as a configuration item ... if we do that it would make all this go away as these events mostly wouldn't be indexed going forward unless someone explicitly is looking for that in which case we presume they know what they are doing

Reply:
My usual strategy is to have the function emit the events. If the caller doesn't want those events, they can set a new event manager in the context that is provided when the function is called.